scripts/TRN(title matching).R

```diff
@@ -0,0 +1,135 @@
+## Script to compile all TRNs and titles of IV + EU trials to then sort by country/sponsor/etc for title matching
+# Generates pairings of EU and IV trials and their respective titles based on title matching algorithm
+
+##########################################################
+
+library(dplyr)
+library(tidyr)
+library(readr)
+library(here)
+library(fs)
+library(lubridate)
+library(stringr)
+library(stringdist)
+library(ctregistries)
+library(cli)
+
+dir_raw <- here("data", "raw")
+dir_processed <- here("data", "processed")
+
+# Download IV TRNs and titles. German full titles are here but the NCT titles here will need to be replaced by what's in studies.csv
+trials <- read_csv(path(dir_processed, "trials.csv"))
+
+# Download EU trial data dump, includes TRNs and full titles
+EU_dump <- read_csv(path(dir_raw, "euctr_euctr_dump-2024-02-03-054239.csv"))
+
+# Download studies.csv, which includes the 'official title' field for all NCTs
+studies <- read_csv(path(dir_raw,"studies.csv" ))
+
+##########################################################
+
+# extract IDs and Titles, and whatever information we want to filter by here for the EU trials.
+# Since IV trials don't have country info like EU trials, we will just get the ID and title here
+IV_ids <- trials |>
+          select(id,title) |>
+          unique()
+
+EU_ids <- EU_dump |>
+          select(eudract_number,
+                 member_state_concerned,
+                 national_competent_authority,
+                 full_title_of_the_trial)
+
+EU_ids <- rename(EU_ids, id = eudract_number, state = member_state_concerned, title = full_title_of_the_trial, authority = national_competent_authority)
+
+NCT_full_titles <- studies |>
+                  select(nct_id, official_title) |>
+                  rename(id = nct_id) |>
+                  rename(title = official_title)
+
+# Update NCT titles in IV_ids, leaving DRKS titles intact
+IV_ids <- rename(IV_ids, brief = title)
+NCT_full_titles <- rename(NCT_full_titles, official = title)
+
+IV_updated_titles <- left_join(IV_ids, NCT_full_titles, by = "id") |>
+                     mutate(title = ifelse(is.na(official), brief, official)) |>
+                     select(-brief, -official)
+
+# Filter out EU trials which dont have 'Germany' listed in member_state_concerned AND national_competent authority
+# Yields table with 13068 unique trials which mention Germany in both of these fields
+# Can filter however we want here.
+
+EU_only_German <- EU_ids |>
+                  filter(grepl('Germany', state) & grepl('Germany', authority))
```

Comment:
Changed to only searching for Germany in 'state', since state and authority are always the same. I looked into using 'sponsors' as a filter, and interestingly, got a smaller list than when I just used 'state' (13038 vs 12790). I guess there are more than a few trials that mention Germany in their 'state' field and not in their 'sponsors' field.